sycl/include/sycl/properties/accessor_properties.hpp

```diff
@@ -18,7 +18,10 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace property {
 
-class no_init : public detail::DataLessProperty<detail::NoInit> {};
+class no_init : public detail::DataLessProperty<detail::NoInit> {
+public:
+  bool operator==(const no_init &) const { return true; }
```

Comment:
Could you please explain what is the reason to have this method implemented?

Reply:
https://github.com/intel/llvm-test-suite/pull/1165/files#diff-d9e7fb7e83aab4b5f4c09b67bf41ee6595df64b9bdc97f04163d7332e2651c98R760 we need it just for E2E test. I can remove it if you think we don't really need it.

Reply:
The test uses API which is not defined by the spec [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_properties_2) and therefore I wouldn't consider it a good E2E test.

You should probably use `static_assert` in combination with `std::is_same`

Reply:
Removed this operator [6d19248](https://github.com/intel/llvm/pull/6614/commits/6d1924846ce72d04b390a4fcc3ecfcd028274eef)